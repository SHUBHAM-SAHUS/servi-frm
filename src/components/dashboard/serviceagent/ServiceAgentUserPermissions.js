import React from 'react';
import { reduxForm, Field } from 'redux-form';
import { validate } from '../../../utils/Validations/roleValidation';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { CustomCheckbox } from '../../../components/common/customCheckbox';
import { Upload, Icon, Menu, Dropdown, Progress, Select, Radio } from 'antd';
import * as actions from '../../../actions/organizationUserAction';
import { Strings } from '../../../dataProvider/localize';

class ServiceAgentUserPermissions extends React.Component {
  render() {
    return (
      <div className="col-md-3">
        {(this.props.userId === undefined || this.props.userId === false) ?
          <div className="sf-card">

            <div className="sf-card-head abb-1 d-flex justify-content-between align-items-center">
              <h4 className="sf-sm-hd sf-pg-heading">{Strings.perm_title}</h4>
            </div>

            <div className="sf-card-body">
              <div className="user-p-notxt d-flex justify-content-between">
                <img src="/images/owl-img.png" />
                <span>{Strings.perm_info_text}</span>
              </div>
            </div>

          </div> :

          <div className="sf-card mb-4">
            <form>
              <div className="sf-card-head abb-1 d-flex justify-content-between align-items-center">
                <h4 className="sf-xs-hd sf-pg-heading">{Strings.perm_card_title}
                  <span className="uhd-name">(Alex Smith)</span></h4>
              </div>

              <div className="sf-card-body">
                <div className="user-mng-chkbx">
                  <div className="sf-checkbox-b">
                    <div className="sf-roles-group">
                      <Radio.Group onChange={this.handleSizeChange}>
                        <Radio.Button value="bdm"><span className="rbp-btn">BDM</span></Radio.Button>
                        <Radio.Button value="accountManagert"><span className="rbp-btn">Account Manager</span></Radio.Button>
                        <Radio.Button value="administrator"><span className="rbp-btn">Administrator</span></Radio.Button>
                        <Radio.Button value="financeManager"><span className="rbp-btn">Finance Manager</span></Radio.Button>
                        <Radio.Button value="staff"><span className="rbp-btn">{Strings.role_staff}</span></Radio.Button>
                      </Radio.Group>
                    </div>
                  </div>

                  <div className="more-role-prmison">
                    <div className="sf-chkbx-group sf-updt-roles sf-scroll-bar h-400">
                      <Field name="read_only" label={Strings.perm_scope_doc} component={CustomCheckbox} />
                      <Field name="modify" label={Strings.perm_job_book} component={CustomCheckbox} />
                      <Field name="create" label={Strings.perm_generate_quote} component={CustomCheckbox} />
                      <Field name="delete" label={Strings.perm_send_quote} component={CustomCheckbox} />
                      <Field name="all" label={Strings.perm_generate_doc} component={CustomCheckbox} />
                    </div>
                  </div>

                </div>
                <div className="all-btn multibnt mt-3">
                  <div className="btn-hs-icon d-flex justify-content-between">
                    <button onClick={this.handleCancel} className="bnt bnt-normal">
                      {Strings.cancel_btn}</button>
                    <button type="submit" className="bnt bnt-active">
                      {Strings.update_btn}</button>
                  </div>
                </div>

              </div>
            </form>
          </div>}

      </div>
    );
  }
}

export default ServiceAgentUserPermissions;