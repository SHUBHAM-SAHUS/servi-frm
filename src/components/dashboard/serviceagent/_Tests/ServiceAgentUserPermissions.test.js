import React from 'react';
import { shallow } from 'enzyme';
import ServiceAgentUserPermissions from '../ServiceAgentUserPermissions';

describe('ServiceAgentUserPermissions Component Test', () => {

    var container
    beforeEach(() => {
        container = shallow(<ServiceAgentUserPermissions />)
    })

    it('should render component service agent user permissions correctly', () => {

        expect(container).toMatchSnapshot();
    });

    it('Render the connected(SMART) ServiceAgentUserPermissions component', () => {
        expect(container.length).toEqual(1)
    });

})

