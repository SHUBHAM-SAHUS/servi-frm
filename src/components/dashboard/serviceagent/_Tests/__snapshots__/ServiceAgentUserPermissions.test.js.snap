// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceAgentUserPermissions Component Test should render component service agent user permissions correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServiceAgentUserPermissions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="sf-card"
      >
        <div
          className="sf-card-head abb-1 d-flex justify-content-between align-items-center"
        >
          <h4
            className="sf-sm-hd sf-pg-heading"
          >
            User Permissions
          </h4>
        </div>
        <div
          className="sf-card-body"
        >
          <div
            className="user-p-notxt d-flex justify-content-between"
          >
            <img
              src="/images/owl-img.png"
            />
            <span>
              Permissions are visible when a use is selected from the users panel
            </span>
          </div>
        </div>
      </div>,
      "className": "col-md-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sf-card-head abb-1 d-flex justify-content-between align-items-center"
          >
            <h4
              className="sf-sm-hd sf-pg-heading"
            >
              User Permissions
            </h4>
          </div>,
          <div
            className="sf-card-body"
          >
            <div
              className="user-p-notxt d-flex justify-content-between"
            >
              <img
                src="/images/owl-img.png"
              />
              <span>
                Permissions are visible when a use is selected from the users panel
              </span>
            </div>
          </div>,
        ],
        "className": "sf-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="sf-sm-hd sf-pg-heading"
            >
              User Permissions
            </h4>,
            "className": "sf-card-head abb-1 d-flex justify-content-between align-items-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "User Permissions",
              "className": "sf-sm-hd sf-pg-heading",
            },
            "ref": null,
            "rendered": "User Permissions",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="user-p-notxt d-flex justify-content-between"
            >
              <img
                src="/images/owl-img.png"
              />
              <span>
                Permissions are visible when a use is selected from the users panel
              </span>
            </div>,
            "className": "sf-card-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  src="/images/owl-img.png"
                />,
                <span>
                  Permissions are visible when a use is selected from the users panel
                </span>,
              ],
              "className": "user-p-notxt d-flex justify-content-between",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "/images/owl-img.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Permissions are visible when a use is selected from the users panel",
                },
                "ref": null,
                "rendered": "Permissions are visible when a use is selected from the users panel",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="sf-card"
        >
          <div
            className="sf-card-head abb-1 d-flex justify-content-between align-items-center"
          >
            <h4
              className="sf-sm-hd sf-pg-heading"
            >
              User Permissions
            </h4>
          </div>
          <div
            className="sf-card-body"
          >
            <div
              className="user-p-notxt d-flex justify-content-between"
            >
              <img
                src="/images/owl-img.png"
              />
              <span>
                Permissions are visible when a use is selected from the users panel
              </span>
            </div>
          </div>
        </div>,
        "className": "col-md-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sf-card-head abb-1 d-flex justify-content-between align-items-center"
            >
              <h4
                className="sf-sm-hd sf-pg-heading"
              >
                User Permissions
              </h4>
            </div>,
            <div
              className="sf-card-body"
            >
              <div
                className="user-p-notxt d-flex justify-content-between"
              >
                <img
                  src="/images/owl-img.png"
                />
                <span>
                  Permissions are visible when a use is selected from the users panel
                </span>
              </div>
            </div>,
          ],
          "className": "sf-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4
                className="sf-sm-hd sf-pg-heading"
              >
                User Permissions
              </h4>,
              "className": "sf-card-head abb-1 d-flex justify-content-between align-items-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User Permissions",
                "className": "sf-sm-hd sf-pg-heading",
              },
              "ref": null,
              "rendered": "User Permissions",
              "type": "h4",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="user-p-notxt d-flex justify-content-between"
              >
                <img
                  src="/images/owl-img.png"
                />
                <span>
                  Permissions are visible when a use is selected from the users panel
                </span>
              </div>,
              "className": "sf-card-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    src="/images/owl-img.png"
                  />,
                  <span>
                    Permissions are visible when a use is selected from the users panel
                  </span>,
                ],
                "className": "user-p-notxt d-flex justify-content-between",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "/images/owl-img.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Permissions are visible when a use is selected from the users panel",
                  },
                  "ref": null,
                  "rendered": "Permissions are visible when a use is selected from the users panel",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
